Terminal.Gui/Drawing/Color.cs

```diff
@@ -110,51 +132,140 @@ public TrueColor (int red, int green, int blue)
+		public static bool TryParse (string text, [NotNullWhen(true)] out TrueColor? trueColor)
+		{
+			// empty color
+			if ((text == null) || (text.Length == 0)) {
+				trueColor = null;
+				return false;
+			}
 
-								CalculateDistance (k.Key, this)
-							));
+			// #RRGGBB or #RGB
+			if ((text [0] == '#') &&
+			    ((text.Length == 7) || (text.Length == 4))) {
+				if (text.Length == 7) {
+					var r = Convert.ToInt32 (text.Substring (1, 2), 16);
+					var g = Convert.ToInt32 (text.Substring (3, 2), 16);
+					var b = Convert.ToInt32 (text.Substring (5, 2), 16);
+					trueColor = new TrueColor (r, g, b);
+				} else {
+					var rText = char.ToString (text [1]);
+					var gText = char.ToString (text [2]);
+					var bText = char.ToString (text [3]);
+
+					var r = Convert.ToInt32 (rText + rText, 16);
+					var g = Convert.ToInt32 (gText + gText, 16);
+					var b = Convert.ToInt32 (bText + bText, 16);
+					trueColor = new TrueColor (r, g, b);
+				}
+				return true;
+			}
+
+			// rgb(XX,YY,ZZ)
+			var match = Regex.Match (text, @"rgb\((\d+),(\d+),(\d+)\)");
+			if (match.Success) {
+				var r = int.Parse (match.Groups [1].Value);
+				var g = int.Parse (match.Groups [2].Value);
+				var b = int.Parse (match.Groups [3].Value);
+				trueColor =  new TrueColor (r, g, b);
+				return true;
+			}
 
-			// get the closest
-			var match = distances.OrderBy (t => t.Item2).First ();
-			return trueColorMap [match.Item1];
+			trueColor = null;
+			return false;
+		}
+
+		/// <summary>
+		/// Converts a <see cref="Color"/> to a <see cref="TrueColor"/> using a default mapping.
+		/// </summary>
+		/// <param name="consoleColor">The <see cref="Color"/> to convert.</param>
+		/// <returns></returns>
+		public static TrueColor FromConsoleColor (Color consoleColor)
+		{
+			switch (consoleColor) {
+			case Color.Black: return new TrueColor (0, 0, 0);
+			case Color.Blue: return new TrueColor (0, 0, 0x80);
+			case Color.Green: return new TrueColor (0, 0x80, 0);
+			case Color.Cyan: return new TrueColor (0, 0x80, 0x80);
+			case Color.Red: return new TrueColor (0x80, 0, 0);
+			case Color.Magenta: return new TrueColor (0x80, 0, 0x80);
+			case Color.Brown: return new TrueColor (0xC1, 0x9C, 0x00); // TODO confirm this
+			case Color.Gray: return new TrueColor (0xC0, 0xC0, 0xC0);
+			case Color.DarkGray: return new TrueColor (0x80, 0x80, 0x80);
+			case Color.BrightBlue: return new TrueColor (0, 0, 0xFF);
+			case Color.BrightGreen: return new TrueColor (0, 0xFF, 0);
+			case Color.BrightCyan: return new TrueColor (0, 0xFF, 0xFF);
+			case Color.BrightRed: return new TrueColor (0xFF, 0, 0);
+			case Color.BrightMagenta: return new TrueColor (0xFF, 0, 0xFF);
+			case Color.BrightYellow: return new TrueColor (0xFF, 0xFF, 0);
+			case Color.White: return new TrueColor (0xFF, 0xFF, 0xFF);
+			default: return new TrueColor (0x80, 0, 0); // TODO getting unknown color value (-1) during JSON serialization
```

Comment:
I was getting an error here when loading the UI catalog. There are some colors that were loading with a -1 value. Since the serialization stuff looks new wasn't sure what the expected behavior is.

Reply:
Added the relevant code from the Converter example you linked. Flipped back to throwing an Exception and the UICatalog loads fine now.